src/WebJobs.Script/Host/Kubernetes/KubernetesDistributedLockManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    // This is an implementation of IDistributedLockManager to be used when running
+    // in Kubernetes environments.
+    internal class KubernetesDistributedLockManager : IDistributedLockManager
+    {
+        private readonly ILogger _logger;
+        private readonly KubernetesClient _kubernetesClient;
+
+        public KubernetesDistributedLockManager(ILoggerFactory loggerFactory, IEnvironment environment)
+        {
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+            _kubernetesClient = new KubernetesClient(environment);
+        }
+
+        public async Task<string> GetLockOwnerAsync(string account, string lockId, CancellationToken cancellationToken)
+        {
+            var response = await _kubernetesClient.GetLock(lockId);
+            return response.Owner;
+        }
+
+        public async Task ReleaseLockAsync(IDistributedLock lockHandle, CancellationToken cancellationToken)
+        {
+            var kubernetesLock = (KubernetesLockHandle)lockHandle;
+            var response = await _kubernetesClient.ReleaseLock(kubernetesLock.LockId, kubernetesLock.Owner);
+            response.EnsureSuccessStatusCode();
+        }
+
+        public async Task<bool> RenewAsync(IDistributedLock lockHandle, CancellationToken cancellationToken)
+        {
+            var kubernetesLock = (KubernetesLockHandle)lockHandle;
+            await _kubernetesClient.TryAcquireLock(kubernetesLock.LockId, kubernetesLock.Owner, TimeSpan.FromSeconds(Convert.ToDouble(kubernetesLock.LockPeriod)), cancellationToken);
+            return true;
```

Comment:
Should `return true;` here be also conditional on `if (string.IsNullOrEmpty(kubernetesLock.LockId))`? The lock implementation will technically auto-renew, but in case the auto-renew fails for some reason and someone else grabs the lock, calling renew on the lock here should return false I think.

Reply:
+1 . This should return true only Acquiring lock succeeds